src/Inventory.zig

```diff
@@ -1414,6 +1569,58 @@ pub const Command = struct { // MARK: Command
 			};
 		}
 	};
+
+	const AddHealth = struct { // MARK: AddHealth
+		source: ?*main.server.User,
+		health: f32,
+		cause: main.game.DamageType,
+		previous: f32,
+
+		pub fn run(self: AddHealth, allocator: NeverFailingAllocator, cmd: *Command, side: Side, _: ?*main.server.User, _: Gamemode) error{serverFailure}!void {
+			if (side == .server) {
+				if (self.source.?.gamemode.raw == .creative) return;
+
+				cmd.executeBaseOperation(allocator, .{.addHealth = .{
+					.health = self.health,
+					.cause = self.cause,
+					.previous = self.previous,
+					.source = self.source,
+				}}, side);
+			} else {
+				if (main.game.Player.gamemode.raw == .creative) return;
+
+				cmd.executeBaseOperation(allocator, .{.addHealth = .{
+					.health = self.health,
+					.cause = self.cause,
+					.previous = self.previous,
+					.source = null,
+				}}, side);
+			}
```

Comment:
Do we really need to branch over the base operation?
Shouldn't self.source already be null on the client side?